src/plots/cartesian/axes.js

```diff
@@ -2918,6 +2919,7 @@ axes.drawGrid = function(gd, ax, opts) {
     grid.attr('transform', opts.transFn)
         .attr('d', opts.path)
         .call(Color.stroke, ax.gridcolor || '#ddd')
+        .style('stroke-dasharray', Drawing.dashStyle(ax.griddash, ax.gridwidth))
```

Comment:
@alexcjohnson with this implementation on master (which is similar to what was used in `carpet`), changing `gridwidth` also changes the height of `dash` lines. Here is a [codepen](https://codepen.io/MojtabaSamimi/pen/MWrOWRm?editors=0010) illustrating it.
Is it the effect that we want? 
Perhaps we shouldn't pass the optional `ax.gridwidth` to `Drawing.dashStyle`.

Reply:
Thanks for bringing this up @archmoj - I think it is what we want, it's consistent with how `line.dash` works in scatter traces: the named dash patterns scale their lengths with the line width, but if you don't want that you can specify the dash pattern explicitly as a list of pixel lengths (the simplest being a single length like `'5px'` if you just want to alternate that length on and off).